lms/djangoapps/teams/views.py

```diff
@@ -0,0 +1,508 @@
+            only through merge patch.
+
+        **Example Requests**:
+
+            GET /api/team/v0/teams/{team_id}}
+
+            PATCH /api/team/v0/teams/{team_id} "application/merge-patch+json"
+
+        **Response Values for GET**
+
+            If the user is logged in, the response contains the following fields:
+
+                * id: The team's unique identifier.
+
+                * name: The name of the team.
+
+                * is_active: True if the team is currently active. If false, the team
+                  is considered "soft deleted" and will not be included by default in
+                  results.
+
+                * course_id: The identifier for the course this team belongs to.
+
+                * topic_id: Optionally specifies which topic the team is
+                  associated with.
+
+                * date_created: Date and time when the team was created.
+
+                * description: A description of the team.
+
+                * country: Optionally specifies which country the team is
+                  associated with.
+
+                * language: Optionally specifies which language the team is
+                  associated with.
+
+                * membership: A list of the users that are members of the team. See
+                  membership endpoint for more detail.
+
+            For all text fields, clients rendering the values should take care
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If the user is not logged in or is not staff for the course, a 403 error
+            is returned.
+
+            If the specified team does not exist, a 404 error is returned.
+
+        **Response Values for PATCH**
+
+            Only staff can patch teams. If the specified team does not exist, a
+            404 error is returned. If the user is not staff, a 403 error is
+            returned.
+
+            If "application/merge-patch+json" is not the specified content type,
+            a 415 error is returned.
+
+            If the update could not be completed due to validation errors, this
+            method returns a 400 error with all error messages in the
+            "field_errors" field of the returned JSON.
+    """
+
+    class IsEnrolledOrIsStaff(permissions.BasePermission):
+        """Permission that checks to see if the user is enrolled in the course or is staff."""
+
+        def has_object_permission(self, request, view, obj):
+            """Returns true if the user is enrolled or is staff."""
+            return CourseEnrollment.is_enrolled(request.user, obj.course_id) or request.user.is_staff
+
+    authentication_classes = (SessionAuthentication, OAuth2Authentication)
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrReadOnly, IsEnrolledOrIsStaff,)
```

Comment:
I don't understand why the "IsStaffOrReadOnly" permission class is still here. It is superseded by IsEnrolledOrIsStaff, isn't it?